### ISSUE TITLE
annotation rule does not detect violations when array-syntax annotations are on the same line as other annotations

### ISSUE THREAD

**Issue by shashachu** [pinterest/ktlint#1765]
## Steps to Reproduce
```
@[JvmStatic Provides] @Foo
fun foo() = 42
```

is correctly reformatted to

```
@[JvmStatic Provides]
@Foo
fun foo() = 42
```

in IJ, but the `annotation` rule doesn't detect any violations.

## Your Environment

* Version of ktlint used: 0.48.1
